docarray/doc_index/abstract_doc_index.py

```diff
@@ -342,14 +344,22 @@ def __getitem__(
             return_singleton = False
         # retrieve data
         doc_sequence = self._get_items(key)
+        # check data
+        if return_singleton and len(doc_sequence) == 0:
+            raise KeyError(f'No document with id {key} found')
+
         # cast output
         if isinstance(doc_sequence, DocumentArray):
             out_da: DocumentArray[TSchema] = doc_sequence
         else:
+            if type(doc_sequence) is Sequence[Dict]:
```

Comment:
We cannot use `issubclass` or `isinstance` between `doc_sequence` and `Sequence[Dict]`. It causes `TypeError: Subscripted generics cannot be used with class and instance checks`.

We can check change to `if instance(doc_sequence[0], dict)`, since we already check length above.